kubernetes/cache/cache.go

```diff
@@ -149,6 +151,34 @@ func NewKialiCache() (KialiCache, error) {
 	kialiCacheImpl.k8sApi = istioClient.GetK8sApi()
 	kialiCacheImpl.istioApi = istioClient.Istio()
 
+	// Update SA Token
+	ticker := time.NewTicker(tokenExpireDuration)
+	quit := make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				if newToken, err := kubernetes.GetKialiToken(); err != nil {
+					log.Errorf("Error updating Kiali Token %v", err)
+				} else {
+					log.Debug("Kiali Cache: Updating token")
+					istioConfig.BearerToken = newToken
+					var errorInitClient error
+					istioClient, errorInitClient = kubernetes.NewClientFromConfig(&istioConfig)
+					if errorInitClient != nil {
+						log.Errorf("Error creating new Client From Config %v", errorInitClient)
+					}
+					kialiCacheImpl.istioClient = *istioClient
+					kialiCacheImpl.k8sApi = istioClient.GetK8sApi()
+					kialiCacheImpl.istioApi = istioClient.Istio()
```

Comment:
We shouldn't try to recreate the cache every 60 seconds. Can we read the token every 60 seconds to check for updates but only expire if there's been some change?

Reply:
- I still need to test the first point. 
- Great idea. I just refresh the cache in case the token has changed.

Reply:
I was able to reproduce the error with the steps provided by @israel-hdez, thanks! 

As @nrfox it seem it was an issue with namespace cached. After the last change, we can see the logs: 

```
022-08-31T15:28:17Z INF Kiali Cache: Updating cache with new token
2022-08-31T15:28:17Z INF Adding a RegistryRefreshHandler for [default]
2022-08-31T15:28:17Z INF Waiting for Kiali cache for [namespace: default] to sync
2022-08-31T15:28:17Z INF Kiali cache for [namespace: default] stopped
2022-08-31T15:28:17Z INF Kiali cache for [namespace: default] started
2022-08-31T15:28:17Z INF Adding a RegistryRefreshHandler for [kiali]
2022-08-31T15:28:17Z INF Waiting for Kiali cache for [namespace: kiali] to sync
2022-08-31T15:28:17Z INF Kiali cache for [namespace: kiali] stopped
2022-08-31T15:28:18Z INF Kiali cache for [namespace: kiali] started
```

Instead of the errors